### DISCUSSION TITLE
Хроники Миртаны: Архолос [ОффТоп]

### DISCUSSION THREAD

**Discussion by Kor-Angar** [dosinabox/g2nr_unofficial_update#422] [General]
сабж

**Comment by Kor-Angar** [comment]
По просьбам трудящихся сделал сейв сброней. возможно тебе тоже будет интересно взглянуть :-)

[savegame3.zip](https://github.com/dosinabox/g2nr_unofficial_update/files/7777210/savegame3.zip)

**Reply by Kor-Angar** [reply]
забавно что к некоторым доспехам прикручены анимации

**Comment by dosinabox** [comment]
Я рада, что ты проводишь в этом моде много времени)

**Reply by Kor-Angar** [reply]
интересное высказывание :-) но контекст не совсем понятен :-)

**Reply by dosinabox** [reply]
Ты всегда недолюбливал моды, а тут как в новую часть Готики играешь, с удовольствием, с первыми впечатлениями

**Reply by Kor-Angar** [reply]
я  и щас их недолюбливаю. когда берут Готику и начинают в ней реализовывать какие то странные мысли, внедряя в мир того чего нет. а тут полноценная игра. я не считаю это модом. свой мир, свои правила. а вселенная готики. и всё нормально. даже в плане баланса, решили ловкостное оружие сделать? ну давайте попробуем. бонусы не влияют на прокачку? интересно посмотреть что из это выйдет.
очень много конечно уже замечаний, но они абсолютно не влияют на общее впечатление. 

мне вот уже хочется побыстрее доиграть что бы начать заново. хочу попробовать двуручное оружие с ловкостью. чисто по приколу, посмотреть. в принципе уже накопленных знаний достаточно, но надо всё таки хотя бы четвертую главу пройти что бы заново начать :-)

**Comment by Kor-Angar** [comment]
смотри какая интересная фича https://www.youtube.com/watch?v=IEjoRYPR6Uo

**Reply by Kor-Angar** [reply]
тут конечно пример не самый удачный, но общая концепция я думаю понятна :-)

**Reply by Kor-Angar** [reply]
можно потестить за фермой , там есть и варги и нежить :-) ITSC_TRFWARG

**Reply by dosinabox** [reply]
В оригинале Г2НВ тоже так можно, вроде. Но в обновлении это "исправили".

**Reply by Kor-Angar** [reply]
> В оригинале Г2НВ тоже так можно, вроде. 

ни разу не слышал ничего такого. ну и сам не встречал. это же целая механика игровая: можно стать "лидером" стаи и повести её в атаку куда нибудь. при этом опыт начисляют если даже убивает не герой.

**Reply by dosinabox** [reply]
> - Волки не реагируют на драку с другими волками, кабаны не присоединяются к ГГ в облике волка, убившего их сородича (спасибо ElderGamer).
> - Присоединенные волки больше не умирают после окончания времени жизни вызванных существ. После возвращения ГГ в облик человека они становятся враждебными (спасибо Desomorphine).

**Reply by dosinabox** [reply]
Механика интересная конечно, надо будет изучить подробнее

**Reply by Kor-Angar** [reply]
прикольно. не знал. надо будет попробовать. в классик это наверное есть?

"стайные" хищники?
* волка
* варги
* снепперы

**Reply by Kor-Angar** [reply]
может авторы мода сами не знали про эту фичу :-)

**Reply by Kor-Angar** [reply]
> Волки не реагируют на драку с другими волками, кабаны не присоединяются к ГГ в облике волка, убившего их сородича (спасибо ElderGamer).

тут всё понятно. кабанов в НВ не было и видимо их "кланировали" с волков ни учтя при этом подобной ситуации :-)

**Comment by Kor-Angar** [comment]
Прошёл я наконец Архолос. 6 глава была интересная :-) но мысль не оригинальная конечно. но реализовано нормально и местами душевно.

в целом конечно мод прекрасен, работа организована и поставлена правильна и несмотря на местами слабую реализацию в том числе скриптов, общий результат положительный.

из минусов на мой взгляд это 
а) баланс (игнорирование бонусов при прокачке + гипертрофированная роль перманентных бонусов).
б) всевозможные ограничения (нельзя кого то добить, нельзя открыть дверь(которую герой потом пинком открывает), всякие бессмертные крестьяне, нельзя что то выкинуть, нельзя куда то попасть. и этого всего слишком много в игре. что совсем не в стиле готики :-)
в) слабый основной сюжет
г) тоскания героя за ухо по сюжету

***

но в целом эти минусы никак не портят общего впечатления от мода, так как например тот же мир (дизайн-уровней это называется?) сделан прекрасно, всякие мелкие квесты сделаны прекрасны, гора [кат-сцен](https://www.youtube.com/playlist?list=PLqe88GB0Po9xdT1K6syjJuiY_4avnaAtc), и даже [секретные видосики ](https://www.youtube.com/playlist?list=PLqe88GB0Po9yBFJ1d9qeKBawK3mBL2B8D). Опять же разнообразная еда и крафт оружия. 
ремесло в плане баланса влияет сильнее чем фракция - прекрасное решение. 

Польская Озвучка  прекрасная. Мне тут в Вк "современные" игроки пишут почти ежедневно про русскую озвучку, то им робота подавай, то какие то энтузиасты переведут, но это всё конечно будет недостижимо. Ибо игра актёров это игра актёров. а студийная запись это студийная запись, а сведение всего это сведения всего :=)

Музыка прекрасна. 

***
В принципе хорошо, что я накануне заморочился с локализацией для Готики 2 классик(неиначе как спящий во сне нашептал).

Так как играя в архолос без проблем через парсер подключал фрагменты локализаций вплоть до субтитров, ни прикасаясь к скриптам игры. Это прям хорошо. 

и твои/ мои наработки в работе со статисткой тоже пригодились, так как через парсер  я тоже быстро организовал книгу со статистикой что позволило многое сравнивать ни в ощущениях, а в сухих цифрах :-)

***
теперь можно с новыми силами будет вернуть к игре в Готику :-)

**Reply by Kor-Angar** [reply]
еще хорошо в архолос что в него всё так же можно играть без мышки :-)

**Comment by Kor-Angar** [comment]
Архолос  (не-озвучка)

1 https://www.youtube.com/watch?v=K727j9x6l0M
2 https://www.youtube.com/watch?v=8bOIPk9pcp4

**Reply by dosinabox** [reply]
Автоматическая?

**Reply by Kor-Angar** [reply]
вроде нет.
![image](https://user-images.githubusercontent.com/49514764/150701354-a1b26a15-d02c-4f44-9d0b-cfe48b9008d7.png)

**Comment by Kor-Angar** [comment]
Тебе какое оформление "Моряк из Хориниса" больше нравится/понятно?
![image](https://user-images.githubusercontent.com/49514764/150841194-7a710603-ba31-4c88-9a58-e9ac6c07d4c5.png)

**Comment by Kor-Angar** [comment]
чем можно заменить?:-)

```D
{
..
	Info_AddChoice		(Dia_StoryHelper_666, "4.1 ( )",	STORYHELPER_666_CHAPTER4_1);
	Info_AddChoice		(Dia_StoryHelper_666, "4.2 ( )",	STORYHELPER_666_CHAPTER4_2);
	Info_AddChoice		(Dia_StoryHelper_666, "4.3 ( )",	STORYHELPER_666_CHAPTER4_3);
	Info_AddChoice		(Dia_StoryHelper_666, "4.4 ( )",	STORYHELPER_666_CHAPTER4_4);
	Info_AddChoice		(Dia_StoryHelper_666, "4.5 ( )",	STORYHELPER_666_CHAPTER4_5);
	Info_AddChoice		(Dia_StoryHelper_666, "4.6 ( )",	STORYHELPER_666_CHAPTER4_6);
	Info_AddChoice		(Dia_StoryHelper_666, "4.7 ( )",	STORYHELPER_666_CHAPTER4_7);
	Info_AddChoice		(Dia_StoryHelper_666, "4.8 ( )",	STORYHELPER_666_CHAPTER4_8);
	Info_AddChoice		(Dia_StoryHelper_666, "4.9 ( )",	STORYHELPER_666_CHAPTER4_9);
..
};
```


```D
func void STORYHELPER_666_CHAPTER4_1()
{
	STORYHELPER_MAINSTORYLINE_CHAPTER4_EXECUTECOMMON(1);
};
func void STORYHELPER_666_CHAPTER4_2()
{
	STORYHELPER_MAINSTORYLINE_CHAPTER4_EXECUTECOMMON(2);
};
func void STORYHELPER_666_CHAPTER4_3()
{
	STORYHELPER_MAINSTORYLINE_CHAPTER4_EXECUTECOMMON(3);
};
func void STORYHELPER_666_CHAPTER4_4()
{
	STORYHELPER_MAINSTORYLINE_CHAPTER4_EXECUTECOMMON(4);
};
func void STORYHELPER_666_CHAPTER4_5()
{
	STORYHELPER_MAINSTORYLINE_CHAPTER4_EXECUTECOMMON(5);
};
func void STORYHELPER_666_CHAPTER4_6()
{
	STORYHELPER_MAINSTORYLINE_CHAPTER4_EXECUTECOMMON(6);
};

func void STORYHELPER_666_CHAPTER4_7()
{
	STORYHELPER_MAINSTORYLINE_CHAPTER4_EXECUTECOMMON(7);
};
func void STORYHELPER_666_CHAPTER4_8()
{
	STORYHELPER_MAINSTORYLINE_CHAPTER4_EXECUTECOMMON(8);
};
func void STORYHELPER_666_CHAPTER4_9()
{
	STORYHELPER_MAINSTORYLINE_CHAPTER4_EXECUTECOMMON(9);
};
```

**Comment by Kor-Angar** [comment]
в Архолос  помогло лишить проблему с записями в дневник при изменение названия топика :-)

```D
func void b_logentry(var string topic,var string entry)
{
    if(TopicIsMission(topic))
    {
        Log_CreateTopic(topic,LOG_MISSION);
    }
    else
    {
        Log_CreateTopic(topic,LOG_NOTE);
    };
	Log_SetTopicStatus	(topic,LOG_Running);
	b_logentry_old(topic,entry);
	//Log_AddEntry(topic,entry);
    //PrintScreen(PRINT_NEWLOGENTRY,-1,YPOS_LOGENTRY,FONT_SCREENSMALL,2);
	//Snd_Play("LogEntry");
	B_CheckLog (); //
};
```

**Reply by Kor-Angar** [reply]
правда почему толи там то ли нет функции 	B_CheckLog (); то ли называется она по другому, 
но засчёт того что имя у топика и миссии совпадают это оказалость не проблема
```D
	B_CloseTopic (TOPIC_Q101, MIS_Q101, 0, 6);
	B_CloseTopic (TOPIC_Q102, MIS_Q102, 0, 6);
	B_CloseTopic (TOPIC_Q103, MIS_Q103, 0, 6);
	...
	B_CloseTopic (TOPIC_SQ509, MIS_SQ509, 0, 6);
	B_CloseTopic (TOPIC_SQ510, MIS_SQ510, 0, 6);
//	B_CloseTopic (TOPIC_Q601, MIS_Q601, 0, 6);
//	B_CloseTopic (TOPIC_Q602, MIS_Q602, 0, 6);
```

**Comment by Kor-Angar** [comment]
а такая проверка вызываемая из ZS_Dead допустима? 
```D
	// Тролли
	if Hlp_GetInstanceID(slf) == Hlp_GetInstanceID(BDT_8032_TROLL)
	|| Hlp_GetInstanceID(slf) == Hlp_GetInstanceID(BDT_8033_TROLL)
	|| Hlp_GetInstanceID(slf) == Hlp_GetInstanceID(BDT_8034_TROLL)
	{
		if	Npc_IsDead(BDT_8032_TROLL)
		&&	Npc_IsDead(BDT_8033_TROLL)
		&&  Npc_IsDead(BDT_8034_TROLL)
		{
			b_logentry (ITEMDESC_ITWR_BOUNTY_TROLL_1,LOG_WANTEDPOSTER_TROLL_DONE);
		};
	};
```

**Reply by dosinabox** [reply]
да, а что тебя смущает?

**Reply by Kor-Angar** [reply]
меня ничего:-)  это я так решил проверку организовать, вот и уточняю :-) то есть сначала сделал три проверки, а потом решил всё это объединить в одно.

**Comment by dosinabox** [comment]
Можешь найти в файлах мода модель колчана со стрелами, пожалуйста?

**Reply by Kor-Angar** [reply]
сейчас вспомню свой логин/пароль и поищу :-)

**Reply by Kor-Angar** [reply]
ARROWPACKET

не оно?

[QUIVER_ARROWS_INVENTORYITEM.zip](https://github.com/dosinabox/g2nr_unofficial_update/files/8546812/QUIVER_ARROWS_INVENTORYITEM.zip)
[QUIVER_BOLTS_INVENTORYITEM.zip](https://github.com/dosinabox/g2nr_unofficial_update/files/8546813/QUIVER_BOLTS_INVENTORYITEM.zip)

**Reply by dosinabox** [reply]
оно, спасибо

**Comment by Kor-Angar** [comment]
ну теперь можно оценить уровень пи :-)
[scripts.zip](https://github.com/dosinabox/g2nr_unofficial_update/files/9286670/scripts.zip)

**Reply by Kor-Angar** [reply]
немного раскидал инстанции по папкам
[TheChroniclesOfMyrtana.zip](https://github.com/dosinabox/g2nr_unofficial_update/files/9306841/TheChroniclesOfMyrtana.zip)

**Comment by Kor-Angar** [comment]
я вот не понимаю такие условия тут есть **и**  и есть **или**, а скобками что с чем не выделенно. 


```D
func int DIA_DETLOW_TEACH_CONDITION()
{
	if (LOG_GETSTATUS(MIS_SQ204) == LOG_SUCCESS) || (LOG_GETSTATUS(MIS_QM302) == LOG_SUCCESS) && NPC_KNOWSINFO(OTHER, 73979) || (LOG_GETSTATUS(MIS_SQ416) == LOG_SUCCESS)
	{
		return TRUE;
	};
};
```
@dosinabox  покажи пожалуйста что можно вынести в  `return false; `  то есть какой тут квест обязателен для выполнение?

так можно сказать?
```D
if (LOG_GETSTATUS(MIS_SQ204) != LOG_SUCCESS) 
&& (LOG_GETSTATUS(MIS_QM302) != LOG_SUCCESS) 
{
    return false;
};
```

**Reply by dosinabox** [reply]
Условие кривое, разобрать его сможет только автор. Анализировать бесполезно, потому что абсолютно нечитабельно. Вполне возможна и ошибка.
Откуда скрипты? Декомпиляция?

**Reply by dosinabox** [reply]
Можно Гратта спросить, как движок разрулит такое условие. Я не возьмусь сказать.

**Reply by Kor-Angar** [reply]
> Откуда скрипты? Декомпиляция?

да. марвинхелпером

***


> Анализировать бесполезно, потому что абсолютно нечитабельно. Вполне возможна и ошибка.

тогда наверное проще выписать условия, а потом их собрать с точки зрения здравого смысла :-)

```D
if (LOG_GETSTATUS(MIS_SQ204) != LOG_SUCCESS) 
&& (LOG_GETSTATUS(MIS_QM302) != LOG_SUCCESS) 
{
// герой не выполнял квест с  NPC -->  NPC не "дружит" с  ГГ
    return false;
};

if NPC_KNOWSINFO(OTHER, 73979) 
|| (LOG_GETSTATUS(MIS_SQ416) == LOG_SUCCESS)
{
// герой выполнил квест "зверь Архолоса" или о чём то там поговорил с NPC
    return TRUE;
};
```

**Comment by Kor-Angar** [comment]
StoryHelper если надумаешь поиграть в Архолос :-)
[CM_KA_SH.zip](https://github.com/dosinabox/g2nr_unofficial_update/files/9307622/CM_KA_SH.zip)

**Comment by Kor-Angar** [comment]
шутники APPLE_BONUS = APPLE_BONUS - 25;
```D
func void FOOD_APPLEBONUS()
{
	var int FOOD_APPLEBONUS_PRINT1;
	var int FOOD_APPLEBONUS_PRINT2;

	if SELF.ID == 0
	{
		if (APPLE_BONUS >= 7) && (APPLE_BONUS < 15)
		{
			if FOOD_APPLEBONUS_PRINT1 == FALSE
			{
				FOOD_APPLEBONUS_PRINT1 = TRUE;
				PRINT(PRINT_EAT1);
			};
		} else if (APPLE_BONUS >= 15) && (APPLE_BONUS < 25)
		{
			if FOOD_APPLEBONUS_PRINT2 == FALSE
			{
				FOOD_APPLEBONUS_PRINT2 = TRUE;
				PRINT(PRINT_EAT2);
			};
		} else if APPLE_BONUS >= 25
		{
			PRINT(PRINT_EAT3);
			B_RAISEATTRIBUTE(SELF, ATR_STRENGTH, 1);
			SND_PLAY("LevelUp");
			APPLE_BONUS = APPLE_BONUS - 25;
			FOOD_APPLEBONUS_PRINT1 = FALSE;
			FOOD_APPLEBONUS_PRINT2 = FALSE;
		};
	};
};
```

**Reply by dosinabox** [reply]
А шо? Простое обнуление счетчика, хоть и своеобразное

**Reply by Kor-Angar** [reply]
я думаю тут может быть игровая ситуация когда APPLE_BONUS  не будет равен нулю.

**Comment by Kor-Angar** [comment]
что то мне кажется можно как то проще написать

```D
        var int KOSTEN;
        KOSTEN =  (atrAMOUNT - atr) * (atrKOSTEN-1);        // сколько надо ЛП до порога
        KOSTEN += (AMOUNT - atrAMOUNT + atr) * atrKOSTEN;   // сколько надо ЛП после порога
        return KOSTEN;
```

**Reply by Kor-Angar** [reply]
(atrAMOUNT - atr) * atrKOSTEN - (atrAMOUNT - atr) * 1 + (AMOUNT - atrAMOUNT + atr) * atrKOSTEN
(atrAMOUNT - atr + AMOUNT - atrAMOUNT + atr) * atrKOSTEN - (atrAMOUNT - atr) * 1
(atrAMOUNT + AMOUNT - atrAMOUNT) * atrKOSTEN - (atrAMOUNT - atr) * 1
(AMOUNT) * atrKOSTEN - (atrAMOUNT - atr) * 1
(AMOUNT) * atrKOSTEN - atrAMOUNT + atr
(AMOUNT * atrKOSTEN + atr - atrAMOUNT )

**Reply by Kor-Angar** [reply]
return (AMOUNT * atrKOSTEN + atr - atrAMOUNT );

**Reply by Kor-Angar** [reply]
```D
func int C_AtrAMOUNKOSTENT(var int atrAMOUNT,var int atr,var int AMOUNT)
{
    var int atrKOSTEN;
    
    if      (atrAMOUNT >= 120)  { atrKOSTEN = 5;  }
    else if (atrAMOUNT >= 90)   { atrKOSTEN = 4;  }
    else if (atrAMOUNT >= 60)   { atrKOSTEN = 3;  }
    else if (atrAMOUNT >= 30)   { atrKOSTEN = 2;  }
    else                        { atrKOSTEN = 1;  };
    
    if (atr + AMOUNT >= atrAMOUNT)
    {
        if (atr >= atrAMOUNT)                
        {
            return atrKOSTEN * AMOUNT;
        };
        /*
        var int KOSTEN;
        KOSTEN =  (atrAMOUNT - atr) * (atrKOSTEN-1);        // сколько надо ЛП до порога
        KOSTEN += (AMOUNT - atrAMOUNT + atr) * atrKOSTEN;   // сколько надо ЛП после порога
        return KOSTEN;
        */
        return (atrKOSTEN * AMOUNT) + atr - atrAMOUNT;
    };
 
    return false;
};

func int B_GETLEARNCOSTATTRIBUTE(var C_NPC her, var int ATTRIBUT, var int AMOUNT)
{
    var int atr;    atr = 0;
    
  	if      (ATTRIB == ATR_STRENGTH)    {   atr = her.ATTRIBUTE[ATR_STRENGTH]; }
  	else if (ATTRIB == ATR_DEXTERITY)   {   atr = her.ATTRIBUTE[ATR_DEXTERITY]; }
  	else if (ATTRIB == ATR_MANA_MAX)    {   atr = her.ATTRIBUTE[ATR_MANA_MAX]; };
    
    if (C_AtrAMOUNKOSTENT(120,  atr,AMOUNT) != false)    {   return C_AtrAMOUNKOSTENT(120,atr,AMOUNT); };
    if (C_AtrAMOUNKOSTENT(90,   atr,AMOUNT) != false)    {   return C_AtrAMOUNKOSTENT(90,atr,AMOUNT);  };
    if (C_AtrAMOUNKOSTENT(60,   atr,AMOUNT) != false)    {   return C_AtrAMOUNKOSTENT(60,atr,AMOUNT);  };
    if (C_AtrAMOUNKOSTENT(30,   atr,AMOUNT) != false)    {   return C_AtrAMOUNKOSTENT(30,atr,AMOUNT);  };
    
    return AMOUNT;
};    
```

**Comment by Kor-Angar** [comment]
в папке архоласа стим закачал мод готики 2 классик 
![image](https://user-images.githubusercontent.com/49514764/236313623-6d1a855e-c59f-4d94-bf30-3360c47ee16b.png)

**Reply by dosinabox** [reply]
угу, у них там какой-то контент общий с г2

**Comment by Kor-Angar** [comment]
не знаешь что это? что-то сделанное на лего что вызывается при взятие предмета героем?
```D
func void OCNPCINVENTORY__INSERT_HOOK()
{
	var OCNPCINVENTORY NPCINV;
	var C_NPC NPC;
	var C_ITEM ITEM;

	NPCINV = _^(ECX);
	NPC = _^(NPCINV.OWNER);
	if NPC_ISPLAYER(NPC) == FALSE
	{
		return;
	};
	ITEM = _^(ESI);
	TELEMETRY_TAKEITEM(HLP_GETINSTANCEID(ITEM));
	B_ONITEMTAKEN(ITEM);
};

```

**Reply by dosinabox** [reply]
без понятия, ни о чем не говорит

**Comment by Kor-Angar** [comment]
в архолос гора доспехов. а прототипы не использовали. броня глючила, вот переписал броню с использованием прототипов.

**Reply by Kor-Angar** [reply]
```D
prototype ITAR_KM_LEATHER_L_Prototype(C_Item)
{   
    MAINFLAG = ITEM_KAT_ARMOR;
    FLAGS = 0;
    VALUE = VALUE_ITAR_KM_LEATHER_L;
    WEAR = WEAR_TORSO;
    ON_EQUIP = EQUIP_ITAR_KM_LEATHER;
    ON_UNEQUIP = UNEQUIP_ITAR_KM_LEATHER;
    VISUAL = "ITAR_KM_LEATHER_L.3DS";
    
    VISUAL_CHANGE = "KM_ARMOR_WOODCUTTER.asc";
    //VISUAL_CHANGE = "KM_ARM_ARX_L2.asc";
    
    VISUAL_SKIN = 0;
    MATERIAL = MAT_LEATHER;

    PROTECTION[1] = LEATHERL_PROTBLUNT;
    PROTECTION[2] = LEATHERL_PROTEDGE;
    PROTECTION[6] = LEATHERL_PROTPOINT;
    PROTECTION[3] = LEATHERL_PROTFIRE;
    PROTECTION[5] = LEATHERL_PROTMAGIC;   
   
    INV_ROTX = -90;
    INV_ROTY = 0;
    INV_ROTZ = 0;
};

func void B_ITAR_CM_TEXTCOUNT()
{
    item.description    = item.name;
    item.TEXT[0]        = NAME_Prot_Blunt;      item.COUNT[0] = item.protection [PROT_BLUNT];
    item.TEXT[1]        = NAME_Prot_Edge;       item.COUNT[1] = item.protection [PROT_EDGE];
    item.TEXT[2]        = NAME_Prot_Point;      item.COUNT[2] = item.protection [PROT_POINT];
    item.TEXT[3]        = NAME_Prot_Fire;       item.COUNT[3] = item.protection [PROT_FIRE];
    item.TEXT[4]        = NAME_Prot_Magic;      item.COUNT[4] = item.protection [PROT_MAGIC];
    item.TEXT[5]        = NAME_Value;           item.COUNT[5] = item.value;
};

instance ITAR_KM_LEATHER_L(ITAR_KM_LEATHER_L_Prototype)
{
    NAME = NAMEARMOR_KM_LEATHER_L;
    
    B_ITAR_CM_TEXTCOUNT();
};

instance ITAR_KM_LEATHER_L_TIER2(ITAR_KM_LEATHER_L_Prototype)
{
    NAME = NAMEARMOR_KM_LEATHER_L_TIER2;

    PROTECTION[2] += UPGRADEARMOR;
    PROTECTION[1] += UPGRADEARMOR;
    PROTECTION[6] += UPGRADEARMOR;
    PROTECTION[3] += UPGRADEARMOR;
    PROTECTION[5] += UPGRADEARMOR;
    
    B_ITAR_CM_TEXTCOUNT();
};

instance ITAR_KM_LEATHER_L_TIER3(ITAR_KM_LEATHER_L_Prototype)
{
    NAME = NAMEARMOR_KM_LEATHER_L_TIER3;

    PROTECTION[2] += UPGRADEARMOR * 2;
    PROTECTION[1] += UPGRADEARMOR * 2;
    PROTECTION[6] += UPGRADEARMOR * 2;
    PROTECTION[3] += UPGRADEARMOR * 2;
    PROTECTION[5] += UPGRADEARMOR * 2; 
    
    B_ITAR_CM_TEXTCOUNT();
};
```
